OpenRA.Mods.Common/TraitsInterfaces.cs

```diff
@@ -126,4 +126,17 @@ public interface IPreventsAutoTarget
 	{
 		bool PreventsAutoTarget(Actor self, Actor attacker);
 	}
+
+	[Flags]
+	public enum UncloakType
+	{
+		None = 0,
+		Attack = 1,
+		Move = 2,
+		Unload = 4,
+		Infiltrate = 8,
+		Demolish = 16,
+		Damage = 32,
+		Default = Attack | Unload | Infiltrate | Demolish
```

Comment:
This line inspired me to write http://pastebin.com/raw.php?i=vQEKvGYV for situations where this may exist / be merged in the future.